SetupDataPkg/ConfApp/SetupConf.c

```diff
@@ -996,7 +997,11 @@ SetupConfMgr (
         Status = EFI_SUCCESS;
       } else {
         Print (L"\nCurrent configurations are dumped Below in format of *.SVD:\n");
-        Print (L"\n%a\n", StrBuf);
+        Print (L"\n");
+        for (Index = 0; Index < StrBufSize; Index ++) {
+          Print (L"%c", StrBuf[Index]);

```

Comment:
Is %a truncating the string because it hits `\0` and thinks it is the end of the string? I.e. that `%s` would fail in the same way?

Reply:
Nope, there is a limit on how long you can print with %a (gEfiMdePkgTokenSpaceGuid.PcdUefiLibMaxPrintBufferSize). But I feel it will be difficult to predict the length of xml buffer for this printing case.

Alternatively, we can do print in a loop, so that it can print as many chars as possible in one call.

Reply:
It is moved to print maximal characters per call, with a 0 printed check to make sure we do not get stuck in this case.